.circleci/config.yml

```diff
@@ -3,6 +3,72 @@ orbs:
   ruby: circleci/ruby@0.1.2 
 
 jobs:
+  tools:
+      docker:  
+        - image: cimg/base:2020.01
+      steps:
+        - checkout
+        - run:
+            name: Install python dependencies
+            command: |
+              sudo apt-get update
+              sudo apt-get -y install --no-install-recommends make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+        - restore_cache:
+            name: Restore asdf Tools Cache
+            keys:
+              - tool-versions-{{ checksum ".tool-versions" }}-v2
+        - run:
+            name: Install asdf
+            command: |
+              [ ! -f ".tool-versions" ] && echo "file .tool-versions was not found" && exit 1
+              [ -d ~/.asdf ] || git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.7.4
+              echo 'source ~/.asdf/asdf.sh' >> $BASH_ENV
+        - run:
+            name: Install asdf tools
+            command: |
+              asdf install
+              asdf reshim
+              pip install pre-commit
+        - save_cache:
+            name: Save asdf Tools Cache
+            key: tool-versions-{{ checksum ".tool-versions" }}-v2
+            paths:
+              - ~/.asdf
+        - persist_to_workspace:
+            root: ~/
+            paths:
+              - .asdf
+              - .bashrc
+              - cas-shipit
+  lint:
+    docker:  
+        - image: cimg/base:2020.01
+    steps:
+        - checkout
+        - attach_workspace:
+            at: ~/
+        - run:
+            name: install dependencies
+            command: |
+              sudo apt-get update
+              sudo apt-get -y install zlib1g-dev libpq-dev
+        - restore_cache:
+            name: Restore gems
+            keys:
+              - gem-cache-v2-{{ checksum "Gemfile.lock" }}
+        - run:
+            name: pre-commit
+            command: |
+              source ~/.asdf/asdf.sh 
+              asdf global postgres 11.4
+              asdf reshim 
+              bundle install
+              pre-commit
+        - save_cache:
+            name: Save gems
+            key: gem-cache-v2-{{ checksum "Gemfile.lock" }}
+            paths:
+              - /home/circleci/.asdf/installs/ruby/2.6.4/lib/ruby/gems/2.6.0
```

Comment:
`gem environment gemdir` seems to give us the gem dependencies path.
can we use this command to get the path since the path might be dynamic depends on the ruby version in tool-versions.